backend/service/src/service_provider.rs

```diff
@@ -0,0 +1,81 @@
+use std::sync::Arc;
+
+use repository::{RepositoryError, StorageConnection, StorageConnectionManager};
+
+use crate::{
+    auth::{AuthService, AuthServiceTrait},
+    email::{EmailService, EmailServiceTrait},
+    log_service::{LogService, LogServiceTrait},
+    settings::Settings,
+    universal_codes::UniversalCodesService,
+    user_account::{UserAccountService, UserAccountServiceTrait},
+};
+
+pub struct ServiceProvider {
+    pub connection_manager: StorageConnectionManager,
+    pub email_service: Box<dyn EmailServiceTrait>,
+    pub dgraph_service: Box<UniversalCodesService>,
```

Comment:
Didn't feel worth creating a Trait for this, especially as it needs to be async and that requires a special async trait library. I think we can assume that the test cases can run against a real dgraph database locally with all the universal codes data. In other words, I don't need to mock this...